WcaOnRails/app/views/competitions/_index_form.html.erb

```diff
@@ -0,0 +1,24 @@
+<div class="form-group same-line">
+  Event: <br /<br /><%= select_tag(:event, options_for_select(@events, params[:event])) %>
+</div>
+
+<div class="form-group same-line">
+  Region: <br /<br /><%= select_tag(:region, options_for_select(@regions, params[:region])) %>
+</div>
+
+<div class="form-group same-line">
+  Years: <br /<br /><%= select_tag(:years, options_for_select(@years, params[:years])) %>
+</div>
+
+<div class="form-group same-line">
+  Name, City or Venue: <br /<br /><%= text_field_tag "search" %></td>
+</div>
+
+<%= button_tag(type: 'submit', name: "commit", value: "List", class: "btn btn-info") do %>
```

Comment:
I have never liked how these buttons work. I feel like it's very strange that you have remember which view you're on and click the appropriate button to stay on your same view. I would rather see a single "Search" button, along with plain old links to the map view and the list view. On further thought, I'm thinking it might be cool to do the map in pure javascript as a popup on top of everything. We could use `data-` stuff on the rows of the table and use javascript to get the competition data out of the table. Lets talk about this over gchat a bit.